### PR TITLE
feat(docs): add tutorial for large scale app - part 1

### PR DESCRIPTION
**Pull request by dhmlau**
Original PR: https://github.com/strongloop/loopback-next/pull/2939

Breaking down the above PR into smaller PRs for easier review.  
This PR has the introductory part of the tutorial.  
- I've added all the images required for the tutorial in this PR
- `index.md` will have broken links because we have added the later parts of the tutorial.
- sidebar is not updated because we'd like to get the content ready and then have a final PR to update the sidebar so that it will show up in our docs site.

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [ ] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: docs/site/tutorials/core/1-introduction.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+title: 'Introduction of the application scenario'
+keywords: LoopBack 4.0, LoopBack 4
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/core-tutorial-part1.html
+---
+
+# Introduction
+
+At the core of LoopBack 4, we provide a powerful Inversion of Control and
+Dependency Injection container in TypeScript that serves as the foundation to
```
Comment (raymondfeng) [inline docs/site/tutorials/core/1-introduction.md]: The links to IoC and DI are missing.

File: docs/site/tutorials/core/1-introduction.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+manage various artifacts for scalable Node.js applications. Independent of API
```
Comment (raymondfeng) [inline docs/site/tutorials/core/1-introduction.md]: as the foundation to manage many types of collaborating artifacts as building blocks for complex applications.

File: docs/site/tutorials/core/1-introduction.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+and Microservice capabilities offered by the LoopBack framework, these core
+modules can be used as a platform to develop large scale Node.js projects that
+require flexibility, extensibility, and composability.
```
Comment (raymondfeng) [inline docs/site/tutorials/core/1-introduction.md]: In addition to great API and Microservice capabilities offered by the LoopBack framework, these core modules can be used independently as a platform to develop large scale Node.js projects that require flexibility, extensibility, and composability.

File: docs/site/tutorials/core/1-introduction.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+To help our users leverage the core packages, we create a series of tutorials to
+explain why such features matter and how to use them for your development.
+
+## Greeting scenario
```
Comment (raymondfeng) [inline docs/site/tutorials/core/1-introduction.md]: ## Sample scenario

File: docs/site/tutorials/core/index.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+# Build large scale Node.js projects with LoopBack 4
+
+It's super easy to start developing in Node.js, but challenges will get in the
+way as your project grows with more and more modules, components, developers,
+teams, and releases. If you are building on an open framework or large-scale
+application, you probably have increasing number of **hmm** moments in deciding
+between **Be less opinionated** and **Don't repeat yourself**.
+
+To set foundation to build LoopBack 4, we have developed an extensible and
```
Comment (raymondfeng) [inline docs/site/tutorials/core/index.md]: To set a solid foundation

File: docs/site/tutorials/core/1-introduction.md (hunk 1 above)
Comment (dhmlau) [reply docs/site/tutorials/core/1-introduction.md]: Do you want to add the ones from wikipedia or LB docs?

File: docs/site/tutorials/core/index.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+permalink: /doc/en/lb4/core-tutorial.html
+---
+
+# Build large scale Node.js projects with LoopBack 4
+
+It's super easy to start developing in Node.js, but challenges will get in the
+way as your project grows with more and more modules, components, developers,
+teams, and releases. If you are building on an open framework or large-scale
+application, you probably have increasing number of **hmm** moments in deciding
+between **Be less opinionated** and **Don't repeat yourself**.
```
Comment (nabdelgadir) [inline docs/site/tutorials/core/index.md]: Are these two mutually exclusive?

File: docs/site/tutorials/core/2-architecture.md [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+in building large scale applications with great flexibility, composability, and
+extensibility.
+
+The following key modules are used by the LoopBack framework to offer API and
+Microservice capabilities. They can be used independently as a base platform to
+build large scale applications in TypeScript and Node.js.
+
+- @loopback/metadata
+- @loopback/context
+- @loopback/core
```
Comment (nabdelgadir) [inline docs/site/tutorials/core/2-architecture.md]: Should we provide links to their API docs?

File: docs/site/tutorials/core/1-introduction.md [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+[Inversion of Control](https://loopback.io/doc/en/lb4/Context.html) and
+[Dependency Injection](https://loopback.io/doc/en/lb4/Dependency-injection.html)
+container in TypeScript that serves as the foundation to manage many types of
+collaborating artifacts as building blocks for complex applications. In addition
+to great API and Microservice capabilities offered by the LoopBack framework,
+these core modules can be used independently as a platform to develop large
+scale Node.js projects that require flexibility, extensibility, and
+composability.
+
+To help our users leverage the core packages, we create a series of tutorials to
```
Comment (nabdelgadir) [inline docs/site/tutorials/core/1-introduction.md]: Nitpick: `we create` -> `we created`

File: docs/site/tutorials/core/2-architecture.md (hunk 7 above)
Comment (nabdelgadir) [reply docs/site/tutorials/core/2-architecture.md]: Yeah, I think that's better!

File: docs/site/tutorials/core/index.md (hunk 6 above)
Comment (dhmlau) [reply docs/site/tutorials/core/index.md]: @raymondfeng, since you're the original author of the content, what's your thought? :)

Comment (dhmlau) [reply docs/site/tutorials/core/index.md]: @nabdelgadir, While I don't think those two are mutually exclusive, what would be your suggestion?

Comment (raymondfeng) [reply docs/site/tutorials/core/index.md]: They are not mutually exclusive. It's a balance between covering common tasks and allowing different choices. The separation of `extension point` (common) and `extension` (specific) achieves that.

Comment (nabdelgadir) [reply docs/site/tutorials/core/index.md]: Okay, thanks!

**Comment by dhmlau** [comment]
The drop in code coverage indicated in coveralls is about `packages/http-caching-proxy/src/http-caching-proxy.ts`. Since my PR only has docs changes, so I don't think this is caused by my PR.

**Pull request merged**